vector/src/main/java/im/vector/util/MediaAVScanner.java

```diff
@@ -0,0 +1,196 @@
+     *
+     * @param userId the user id
+     */
+    public MediaAVScanner(String userId) {
+        mUserId = userId;
+    }
+
+    /**
+     * Clear the cached scan results.
+     */
+    public void clearCache() {
+        ContentUtils.deleteDirectory(mMediaAVScannerDirectory);
+        mMediaAVScanByUrl = null;
+
+        // TODO cancel pending scans.
+    }
+
+    /**
+     * Open the scanner cache file.
+     *
+     * @param context the context.
+     */
+    private void openMediaAVScannerDict(Context context) {
+
+        // already checked
+        if (null != mMediaAVScanByUrl) {
+            return;
+        }
+
+        mMediaAVScanByUrl = new HashMap<>();
+
+        try {
+            mMediaAVScannerDirectory = new File(context.getApplicationContext().getFilesDir(), MEDIAAVSCANNER_STORE_FOLDER);
+            mMediaAVScannerDirectory = new File(mMediaAVScannerDirectory, mUserId);
+
+            mMediaAVScannerFile = new File(mMediaAVScannerDirectory, FILENAME.hashCode() + "");
+
+            if (!mMediaAVScannerDirectory.exists()) {
+
+                // create dir tree
+                mMediaAVScannerDirectory.mkdirs();
+            }
+
+            if (mMediaAVScannerFile.exists()) {
+                FileInputStream fis = new FileInputStream(mMediaAVScannerFile);
+                ObjectInputStream ois = new ObjectInputStream(fis);
+                mMediaAVScanByUrl = (HashMap) ois.readObject();
+                ois.close();
+                fis.close();
+            }
+        } catch (Exception e) {
+            Log.e(LOG_TAG, "## openMediaAVScannerDict failed " + e.getMessage());
+        }
+    }
+
+    /**
+     * Get the current scan status for a media.
+     * Trigger a scan if it is not already done.
+     *
+     * @param context   the context.
+     * @param url       the media url.
+     * @param callback  optional async response handler.
+     * @return the current scan status.
+     */
+    public ScanStatus scanEncryptedMedia(Context context, String url, @Nullable ApiCallback<Void> callback) {
+        if (null == mMediaAVScanByUrl) {
+            openMediaAVScannerDict(context);
+        }
+
+        if (TextUtils.isEmpty(url)) {
+            return ScanStatus.UNKNOWN;
+        }
+
+        if (mMediaAVScanByUrl.containsKey(url)) {
+            return mMediaAVScanByUrl.get(url);
+        }
+
+        // TODO trigger the scan, use the callback on result if any
+
+        return ScanStatus.IN_PROGRESS;
+    }
+
+    /**
+     * Get the current scan status for an encrypted media.
+     * Trigger a scan if it is not already done.
+     *
+     * @param context the context.
+     * @param mediaInfo  the encrypted media information.
+     * @param callback   optional async response handler.
+     * @return the current scan status.
+     */
+    public ScanStatus scanEncryptedMedia(Context context, EncryptedFileInfo mediaInfo, @Nullable ApiCallback<Void> callback) {
+        if (null == mMediaAVScanByUrl) {
```

Comment:
Should just contains:
return scanEncryptedMedia(context, mediaInfo.url, callback)